File: lib/data/birthday_storage.dart
import 'package:shared_preferences/shared_preferences.dart';

class BirthdayStorage {
  static const _keyDay = 'birthday_day';
  static const _keyMonth = 'birthday_month';

  static Future<void> save(DateTime date) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setInt(_keyDay, date.day);
    await prefs.setInt(_keyMonth, date.month);
  }

  static Future<DateTime?> load() async {
    final prefs = await SharedPreferences.getInstance();
    final month = prefs.getInt(_keyMonth) ?? 1;
    final day = prefs.getInt(_keyDay) ?? 1;
    return DateTime(2000, month, day);
  }

  static Future<void> clear() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(_keyDay);
    await prefs.remove(_keyMonth);
  }
}

File: android/app/src/main/kotlin/com/example/birthday_progress/data/BirthdayStorage.kt
package com.example.birthday_progress.data

import android.content.Context
import android.content.SharedPreferences
import java.util.Calendar

class BirthdayStorage(context: Context) {
    private val prefs: SharedPreferences = context.getSharedPreferences(
        PREFS_NAME, Context.MODE_PRIVATE
    )

    fun load(): Calendar {
        val day = prefs.getLong(KEY_DAY, 1L).toInt()
        val month = prefs.getLong(KEY_MONTH, 1L).toInt()

        return Calendar.getInstance().apply {
            set(Calendar.YEAR, 2000)
            set(Calendar.MONTH, month)
            set(Calendar.DAY_OF_MONTH, day)
            set(Calendar.HOUR_OF_DAY, 0)
            set(Calendar.MINUTE, 0)
            set(Calendar.SECOND, 0)
            set(Calendar.MILLISECOND, 0)
        }
    }

    companion object {
        private const val PREFS_NAME = "FlutterSharedPreferences"
        private const val KEY_DAY = "flutter.birthday_day"
        private const val KEY_MONTH = "flutter.birthday_month"
    }
}